### ISSUE TITLE
Case (in)sensitive headers with HTTP2

### ISSUE THREAD

**Issue by marcelomd** [nbs-system/naxsi#294]
Hi,

While testing the http2 branch we came upon an issue in the whitelists' tests. Specifically the tests that use cookies, but it may happen with other headers. We're aware this is experimental =)

This is related to #227 and commit `bc2d1c`.

Using cookies as example. There are cases where the case of the word 'cookie' (cookie, Cookie, COOKIE, etc.) matter in the test outcome.

Example:
```nginx
=== Blocks, but should allow
--- main_config
--- http_config
include /workspace/naxsi/naxsi_core.rules;
MainRule "str:foobar" "msg:foobar test pattern" "mz:HEADERS" "s:$SQL:42" id:1999;
--- user_files
>>> another-page
ANOTHER CONTENT
--- config
location / {
	 #LearningMode;
	 SecRulesEnabled;
	 DeniedUrl "/RequestDenied";
	 CheckRule "$SQL >= 8" BLOCK;
	 root $TEST_NGINX_SERVROOT/html/;
	index index.html index.htm;
	 BasicRule wl:1999 "mz:$URL:/another-page|$HEADERS_VAR:cookie";
}
location /RequestDenied {
	 return 412;
}
--- more_headers
COOKIE: foobar
--- request
GET /another-page
--- error_code: 412
```

Is this a bug or am I doing something wrong?
As always, we can help with code, but I'm not sure of the best way to attack this.

Thanks!

**Comment by buixor** [comment]
Hello !

I'm going to look at this probably next week.
However, I want to point out how exemplary this issue is :)

Thanks !

**Comment by marcelomd** [comment]
Hahaha, that's awesome. I just did my homework =)
Glad we can help.

Thanks!

**Comment by buixor** [comment]
Hello,

At first glance, while it's probably a bug, it's not surprising.
From my understanding with http2 (and that's the reason behind the original crashes), "common" header names are stored by nginx in r/o memory (the original SEGV was caused by an attempt to lower-case incoming header's name iirc). Naxsi being case insensitive, I'm not really surprised by this issue. I'll keep you posted on this one.

**Comment by buixor** [comment]
Hello,

Sorry for the delay :)
I pushed to http2 branch a fix that should adress this (and #227) in a better way.
Please provide feeback !

cheers,

**Comment by marcelomd** [comment]
Hi,

It's passing all our tests now. Looks like it works!
I think we can close this.

Thanks a lot!
